src/StarsRatingBar/StarsRatingBar.js

```diff
@@ -0,0 +1,215 @@
+    const { readOnly } = this.props;
+
+    return Object.values(starIndexes).map(ratingValue => {
+      return readOnly
+        ? this._renderReadOnlyModeStar(ratingValue)
+        : this._renderInteractiveModeStar(ratingValue);
+    });
+  };
+
+  _renderReadOnlyModeStar = ratingValue => {
+    const { readOnly, value } = this.props;
+    const { starsRatingBarSize } = this.state;
+
+    const isFilledStar = ratingValue <= value;
+    return (
+      <StarFilledIcon
+        key={ratingValue}
+        data-hook={dataHooks.filledStar}
+        {...styles(
+          'star',
+          { readOnly, filled: isFilledStar, empty: !isFilledStar },
+          this.props,
+        )}
+        size={starRatingBarSizesInPx[starsRatingBarSize]}
+      />
+    );
+  };
+
+  _renderInteractiveModeStar = ratingValue => {
+    const { value } = this.props;
+    const { starsRatingBarSize, hoveredStarIndex } = this.state;
+
+    const isStarHovered = hoveredStarIndex != -1;
+    const isFilledStar = isStarHovered
+      ? ratingValue <= hoveredStarIndex
+      : ratingValue <= value;
+
+    const commonProps = {
+      key: ratingValue,
+      size: starRatingBarSizesInPx[starsRatingBarSize],
+      onMouseEnter: () => this._onMouseEnter(ratingValue),
+      onMouseLeave: () => this._onMouseLeave(),
+      onClick: () => this._onStarIconClick(ratingValue),
+    };
+
+    return isFilledStar ? (
+      <StarFilledIcon
+        {...commonProps}
+        data-hook={dataHooks.filledStar}
+        data-index={ratingValue}
+        {...styles(
+          'star',
+          { filled: true, hovered: isStarHovered },
+          this.props,
+        )}
+      />
+    ) : (
+      <StarIcon
+        {...commonProps}
+        data-index={ratingValue}
+        {...styles('star', { empty: true, hovered: isStarHovered }, this.props)}
+      />
+    );
+  };
+
+  _shouldShowRateCaption = () => {
+    const { readOnly, rateCaptions } = this.props;
+    let shouldShowRateCaption = false;
+
+    if (rateCaptions) {
+      const isValidRateCaption =
+        Array.isArray(rateCaptions) && rateCaptions.length === 5;
+
+      if (readOnly) {
+        throw new Error('Rate caption is not available in read only mode.');
+      } else if (!isValidRateCaption) {
+        throw new Error('Rate caption must be an array of strings at size 5.');
+      }
+      shouldShowRateCaption = true;
+    }
+
+    return shouldShowRateCaption;
+  };
+
+  _renderRateCaption = () => {
+    const { rateCaptions, value } = this.props;
+    const rateCaptionCurrentLabel = value === 0 ? '' : rateCaptions[value - 1];
+
+    return (
+      <Text
+        dataHook={dataHooks.rateCaption}
+        className={styles.rateCaption}
+        ellipsis
+        size="small"
+        weight="bold"
+        secondary
+      >
+        {rateCaptionCurrentLabel}
+      </Text>
+    );
+  };
+
+  render() {
+    const { dataHook } = this.props;
+
+    return (
+      <div data-hook={dataHook}>
```

Comment:
Should have a root class (even if it's empty)
```suggestion
      <div data-hook={dataHook} {...styles('root', {}, { className })}>
```